atom/common/platform_util_win.cc

```diff
@@ -169,33 +170,298 @@ bool OpenExternal(const GURL& url) {
   return true;
 }
 
+
+// Required COM implementation of IFileOperationProgressSink so we can
+// precheck files before deletion to make sure they can be move to the 
+// Recycle Bin.
+class DelFileProgressSink : public IFileOperationProgressSink {
+  public:
+    DelFileProgressSink();
+    ~DelFileProgressSink(void);
+  private:
+	  ULONG STDMETHODCALLTYPE AddRef(void);
+    HRESULT STDMETHODCALLTYPE QueryInterface(REFIID   riid, LPVOID * ppvObj);
+    ULONG STDMETHODCALLTYPE Release( void);
+
+    HRESULT STDMETHODCALLTYPE StartOperations( void);
+    
+    HRESULT STDMETHODCALLTYPE FinishOperations( 
+    /* [in] */ HRESULT hrResult);
```

Comment:
nit: I would personally remove all the IDL annotation (/*[in]*/), it serves no real purpose here and adds noise (especially the ones like [string][unique][in])